tests/nfs_pvc_invalid_backend_sc_test.go

```diff
@@ -0,0 +1,226 @@
+					break
+				}
+				// event will be generated after 60 seconds
+				time.Sleep(time.Second * 10)
+			}
+			Expect(isExpectedEventExist).To(BeTrue(), "ProvisioningFailed event should exist with PVC bound timed out")
+
+			// Add integration test finalizer on backend PVC, so that object
+			// will remains same even after deletion unless removal of finalizer
+			// NOTE: Below snippet will be removed after merging
+			backendPVCObj, err := Client.getPVC(OpenEBSNamespace, backendPVCName)
+			Expect(err).To(BeNil(), "while fetching pvc %s/%s", OpenEBSNamespace, backendPVCName)
+			backendPVCObj.Finalizers = append(backendPVCObj.Finalizers, integrationTestFinalizer)
+			_, err = Client.updatePVC(backendPVCObj)
+			Expect(err).To(BeNil(), "while adding %s finlaizer to pvc %s/%s", integrationTestFinalizer, OpenEBSNamespace, backendPVCName)
+		})
+
+		It("should not export any events to REST service", func() {
+			var backendPVCObj *corev1.PersistentVolumeClaim
+			var err error
+			var isEventReceived bool
+
+			for retries := 0; retries < maxRetryCount; retries++ {
+				backendPVCObj, err = Client.getPVC(OpenEBSNamespace, backendPVCName)
+				Expect(err).To(BeNil(), "while fetching backend pvc %s/%s", OpenEBSNamespace, backendPVCName)
+
+				_, isEventReceived = backendPVCObj.Annotations[nfs.VolumeCreateNFSPVCKey]
+				if isEventReceived {
+					break
+				}
+				// Reconciliation will happen at every 60 seconds but if any error occurs it will
+				// get reconcile easily
+				time.Sleep(time.Second * 10)
+			}
+			Expect(isEventReceived).To(BeFalse(), "REST service shouldn't receive any events")
+			_, isNFSPVCAnnoExist := backendPVCObj.Annotations[nfs.VolumeCreateNFSPVCKey]
+			_, isNFSPVAnnoExist := backendPVCObj.Annotations[nfs.VolumeCreateNFSPVKey]
+			_, isBackendPVCExist := backendPVCObj.Annotations[nfs.VolumeCreateBackendPVCKey]
+			_, isBackendPVExist := backendPVCObj.Annotations[nfs.VolumeCreateBackendPVKey]
+			Expect(isNFSPVCAnnoExist).To(BeFalse(), "REST service shouldn't receive any create event but has NFS pvc name")
+			Expect(isNFSPVAnnoExist).To(BeFalse(), "REST service shouldn't receive any create event but has NFS pv name")
+			Expect(isBackendPVCExist).To(BeFalse(), "REST service shouldn't receive any create event but has backend pvc name")
+			Expect(isBackendPVExist).To(BeFalse(), "REST service shouldn't receive any create event but has backend pv name")
+		})
+	})
+
+	When("pvc "+pvcName+"is deleted", func() {
+		It("should get delete", func() {
+			err := Client.deletePVC(applicationNamespace, pvcName)
+			Expect(err).To(BeNil(), "while deleting pvc %s/%s", applicationNamespace, pvcName)
+		})
+		It("should get delete and shouldn't send any events to REST service", func() {
+			var backendPVCObj *corev1.PersistentVolumeClaim
+			var err error
+			var isDeleteEventReceived bool
+
+			for retries := 0; retries < maxRetryCount; retries++ {
+				backendPVCObj, err = Client.getPVC(OpenEBSNamespace, backendPVCName)
```

Comment:
missing check for non-empty `backendPVCName`

Reply:
Done